src/main/java/org/lisoft/lsml/view_fx/controls/EquippedItemCell.java

```diff
@@ -245,8 +257,9 @@ private VBox makeEngineGraphic(final Engine aEngine) {
                 aEvent.consume();
             }
         });
-        engineXl.setSelected(aEngine.getType() == EngineType.XL);
+        engineType.getSelectionModel().select(aEngine.getType());
         engineRating.getSelectionModel().select(Integer.valueOf(aEngine.getRating()));
+        regenerateEngineRatingDropDown();
```

Comment:
I think passing the current engine as an argument here would simplify the code in regenerateEngineRatingDropDown.

Reply:
Done (also changing changeEngine to return `Optional<Engine>` to make it so that we actually have an `Engine` at all the places where this is called.)